apps/ledger-live-mobile/src/screens/Accounts/index.tsx

```diff
@@ -119,26 +119,33 @@ function Accounts({ navigation, route }: NavigationProps) {
             </Text>
           </Flex>
         )}
-        <List
-          testID="accounts-list"
-          data={flattenedAccounts}
-          renderItem={renderItem}
-          keyExtractor={(i: AccountLike) => i.id}
-          ListHeaderComponent={
-            <Flex mt={3} mb={3}>
-              <Text testID="accounts-list-title" variant="h4">
-                {params?.currencyTicker
-                  ? t("accounts.cryptoAccountsTitle", {
-                      currencyTicker: params?.currencyTicker,
-                    })
-                  : t("accounts.title")}
-              </Text>
-            </Flex>
-          }
-          contentContainerStyle={{
-            paddingHorizontal: 16,
-          }}
-        />
+        {/* If params contains account address or currency, this page will redirect, so show a loading spinner instead */}
+        {params?.address || params?.currency ? (
```

Comment:
```suggestion
        {paramCurrency ? (
```
I had a look where the redirection was occurring and see that it's not dependent on address and also could be written better, suggest to replace the useEffect with something like this:
```
  const paramCurrency = useMemo(
    () => params?.currency && findCryptoCurrencyByKeyword(params.currency.toUpperCase()),
    [params?.currency],
  );

  // Deep linking params redirect
  useEffect(() => {
    if (paramCurrency) {
      const paramAddress = params?.address;
      const account = paramAddress
        ? accounts.find(
            acc =>
              acc.currency.id === paramCurrency.id &&
              acc.freshAddress.toLowerCase() === paramAddress.toLowerCase(),
          )
        : null;

      if (account) {
        navigation.replace(ScreenName.Account, {
          accountId: account.id,
        });
      } else {
        navigation.replace(ScreenName.Asset, {
          currency: paramCurrency,
        });
      }
    }
  }, [paramCurrency, accounts, navigation, params?.address]);
```